apps/jobs/urls.py

```diff
@@ -20,6 +20,11 @@
         views.get_remaining_submissions,
         name="get_remaining_submissions",
     ),
+    url(
+        r'^challenge/(?P<challenge_id>[0-9]+)/'
```

Comment:
Please remove challenge_id and challenge_phase_id from the URL. I think submission ID is sufficient as a submission is related to a phase and phase is related to a challenge in the models.